internal/services/account_sponsorship_service.go

```diff
@@ -113,15 +119,19 @@ func (s *accountSponsorshipService) SponsorAccountCreationTransaction(ctx contex
 		},
 	)
 
-	// TODO: use Channel Accounts instead of using the Distribution Account.
-	distributionAccount, err := s.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: s.SignatureClient.GetDistributionAccountPublicKey()})
+	channelAccountPublicKey, err := s.ChannelAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("getting channel account public key: %w", err)
+	}
```

Comment:
💡 It'd be a good idea to add a custom error for when there is no account available, or no account at all.

Reply:
I think that would be quite useful indeed.